Cloud VM Exposure Remediation Playbooks (#44201)
demisto/content: Packs/CloudSecurityPolicyManagement/ReleaseNotes/1_1_0.md

#### Playbooks

##### New: Azure - VM Instance Network Security Group Remediation

> This playbook adds new Azure Network Security Group (NSG) rules to NSGs attached to a NIC. The new rules will give access only to a private IP address range and block traffic that's exposed to the public internet ([using the private IP of the VM as stated in Azure documentation](https://learn.microsoft.com/en-us/azure/virtual-network/network-security-groups-overview)). For example, if RDP is exposed to the public internet, this playbook adds new firewall rules that only allow traffic from private IP addresses and blocks the rest of the RDP traffic.

Reviewer: ```suggestion
This playbook adds new Azure network security group (NSG) rules to NSGs attached to a NIC. The new rules give access to only a private IP address range and block traffic exposed to the public internet using the private IP of the VM, as explained in the ([Azure documentation](https://learn.microsoft.com/en-us/azure/virtual-network/network-security-groups-overview)). For example, if RDP is exposed to the public internet, this playbook adds new firewall rules that only allow traffic from private IP addresses and blocks the rest of the RDP traffic.
```

Conditions and limitations:

> - Limited to one resource group.

Reviewer: ```suggestion
- Limited to a single resource group.
```

> - Identifies the NSG rule that is exposing the VM to the public internet based on IPv4/IPv6 addresses. Service tags are not supported.

Reviewer: ```suggestion
- Identifies the NSG rule exposing the VM to the public internet via IPv4/IPv6 (service tags are not supported).
```

> - 2 priorities lower than the offending rule priority must be available.

Reviewer: ```suggestion
- Requires at least two available priority slots lower than the offending rule.
```

> - Adds rules to NSGs associated to NICs.

Reviewer: ```suggestion
- Applies only to NSGs associated with NICs.
```

##### New: AWS - EC2 Instance Security Group Remediation

> This playbook identifies the Security Group(s) used on the EC2 interface with the given public IP. It then determines which rules allow access to the interface on the given port and protocol and creates a new copy of the Security Group that has only the identified rules removed. The EC2 interface is then updated to use the modified security group(s). The original Security Groups are left unmodified so that the remediated copy is only applied to the exposed EC2 instance.

Reviewer: ```suggestion
This playbook identifies the security groups used on an EC2 interface with a specific public IP. It determines which rules allow access via the given port and protocol, creates a copy of the security group with those rules removed, and updates the interface to use the modified group. The original security groups remain unchanged, ensuring that only the exposed EC2 instance is updated with the modified security settings.
```

##### New: GCP - VM Instance Firewall Rule Remediation

> This playbook adds new VPC Firewall rules with access only from private IP address range and blocks traffic that’s exposed to public internet. For example, if RDP is exposed this playbook adds new firewall rules that only allows traffic from private IP address and blocks rest of the RDP traffic. New Allow and Block rules created by this playbook target only the exposed asset.

Reviewer: ```suggestion
This playbook adds VPC firewall rules to restrict access to private IP address ranges and block traffic exposed to the public internet. For example, if RDP is exposed this playbook adds new firewall rules that allow traffic only from private IP addresses and block all other RDP traffic. New allow and block rules created by this playbook target only the exposed asset.
```
